src/rpc/mining.cpp

```diff
@@ -267,6 +268,76 @@ static UniValue getmintinginfo(const JSONRPCRequest& request)
+static UniValue getmininginfo(const JSONRPCRequest& request)
+{
+    RPCHelpMan{"getmininginfo",
+        "\nReturns a json object containing mining-related information for all local masternodes",
+        {},
+        RPCResult{
+            "{\n"
+            "  \"blocks\": nnn,             (numeric) The current block\n"
+            "  \"currentblockweight\": nnn, (numeric, optional) The block weight of the last assembled block (only present if a block was ever assembled)\n"
+            "  \"currentblocktx\": nnn,     (numeric, optional) The number of block transactions of the last assembled block (only present if a block was ever assembled)\n"
+            "  \"difficulty\": xxx.xxxxx    (numeric) The current difficulty\n"
+            "  \"networkhashps\": nnn,      (numeric) The network hashes per second\n"
+            "  \"pooledtx\": n              (numeric) The size of the mempool\n"
+            "  \"chain\": \"xxxx\",         (string)  current network name as defined in BIP70 (main, test, regtest)\n"
+            "  \"isoperator\": true|false   (boolean) Local master nodes are available or not \n"
+            "  \"masternodes\": []          (array)   an array of objects which includes each master node information\n"
+            "  \"warnings\": \"...\"        (string)  any network and blockchain warnings\n"
+            "}\n"    
+        },
+        RPCExamples{
+            HelpExampleCli("getmininginfo", "")
+            + HelpExampleRpc("getmininginfo", "")
+        },
+    }.Check(request);
+
+    LOCK(cs_main);
+
+    UniValue obj(UniValue::VOBJ);
+    obj.pushKV("blocks",           (int)::ChainActive().Height());
+    if (BlockAssembler::m_last_block_weight) obj.pushKV("currentblockweight", *BlockAssembler::m_last_block_weight);
+    if (BlockAssembler::m_last_block_num_txs) obj.pushKV("currentblocktx", *BlockAssembler::m_last_block_num_txs);
+    obj.pushKV("difficulty",       (double)GetDifficulty(::ChainActive().Tip()));
+    obj.pushKV("networkhashps",    getnetworkhashps(request));
+    obj.pushKV("pooledtx",         (uint64_t)mempool.size());
+    obj.pushKV("chain",            Params().NetworkIDString());
+
+    // get all masternode operators
+    auto mnIds = pcustomcsview->GetOperatorsMulti();
+    obj.pushKV("isoperator", !mnIds.empty());
+
+    UniValue mnArr(UniValue::UniValue::VARR); // masternodes array
+    for (const auto& mnId : mnIds) {
+        UniValue subObj(UniValue::VOBJ);
+
+        subObj.pushKV("masternodeid", mnId.second.GetHex());
+        CMasternode const & node = *pcustomcsview->GetMasternode(mnId.second);
+        auto state = node.GetState();
+        subObj.pushKV("masternodeoperator", node.operatorAuthAddress.GetHex());// NOTE(sp) : Should this also be encoded? not the HEX
+        subObj.pushKV("masternodestate", CMasternode::GetHumanReadableState(state));
+        auto generate = node.IsActive() && gArgs.GetBoolArg("-gen", DEFAULT_GENERATE);
```

Comment:
Can move `gArgs.GetBoolArg("-gen", DEFAULT_GENERATE);` move outside the loop and assign to a variable.